pkg/minishift/addon/manager/addon_manager.go

```diff
@@ -358,29 +358,36 @@ func setStateAndPriority(addOn addon.AddOn, configMap map[string]*config.AddOnCo
 
 func compareComponentVersions(currentVersion, requiredVersion string, componentName string, versionRange string) error {
 	if strings.HasPrefix(requiredVersion, ">=") {
-		// This will work for both upstream and downstream.
-		if util.VersionOrdinal(currentVersion) < util.VersionOrdinal(strings.TrimPrefix(requiredVersion, ">=")) {
+		currentVersion, _ := semver.Make(currentVersion)
```

Comment:
Is there any reason we are not handling the error here or this is handled somewhere else already?

Reply:
@praveenkumar Here is how I see such least occurrence error. :smile: 
`semver.Make(currentVersion)` is stable enough and even if we get any random string in `currentVersion` or in `requiredVersion`, the equality is definitely going to fail.

I am omitting error checking for the sake of simplicity/readability of code.
Else, three more lines will get added which will break current degree of simplicity/readability.

Hope you understand :) 
Let me know if you have different opinion.

Reply:
`Make func(s string) (Version, error)`  We should handle all errors, because when it comes it is way easier to track it down. So I agree with @praveenkumar to catch the error and handle it.

Reply:
>  Here is how I see such least occurrence error.
semver.Make(currentVersion) is stable enough and even if we get any random string in currentVersion or in requiredVersion, the equality is definitely going to fail.

@budhram what makes `semver.Make` stable enough is returning an error and version parsed value. Now if you put arbitrary version value to parse in `Make` then you don't have any `currentVersion` or `requiredVersion` values to compare and it will go through which is not a definite fail but an unexpected behavior which will be hard to debug.

> I am omitting error checking for the sake of simplicity/readability of code.

You also think about the maintainability of the code.

> Let me know if you have different opinion.

Since version check already handled for required version during metadata parse and for current version in the start time so I am OK here to ignore the error.

Reply:
> so I am OK here to ignore the error.

:+1:

Reply:
> Now if you put arbitrary version value to parse in Make then you don't have any currentVersion or requiredVersion values to compare and it will go through which is not a definite fail but an unexpected behavior which will be hard to debug.

Thats the exact reason we should catch the error and handle it. 


> Since version check already handled for required version during metadata parse and for current version in the start time so I am OK here to ignore the error.

As a general we can not guarantee how  code changes will happen in future. So if because of code change the check gets compromised the wrong version will fall through.